### ISSUE TITLE
Segmentation fault when using --uchime_denovo on a large fasta file (9.3 GB)

### ISSUE THREAD

**Issue by frederic-mahe** [torognes/vsearch#77]
Sorry, it's always during weekends...

That file triggers a segmentation fault during a *de novo* chimera detection. Note that it is not a problem of memory exhaustion, I tried on a node with plenty of memory.
```
gdb ./vsearch-1.1.1-linux-x86_64
GNU gdb (GDB) Red Hat Enterprise Linux (7.2-60.el6)
Copyright (C) 2010 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>...
Reading symbols from vsearch-1.1.1-linux-x86_64...done.
(gdb) run --uchime_denovo neotropical_soil_175_samples_representatives.fas --fasta_width 0 --uchimeout neotropical_soil_175_samples.uchime
Starting program: vsearch-1.1.1-linux-x86_64 --uchime_denovo neotropical_soil_175_samples_representatives.fas --fasta_width 0 --uchimeout neotropical_soil_175_samples.uchime
[Thread debugging using libthread_db enabled]
vsearch v1.1.1_linux_x86_64, 126.0GB RAM, 32 cores
https://github.com/torognes/vsearch

Reading file neotropical_soil_175_samples_representatives.fas 100%  
8789864140 nt in 23735051 seqs, min 32, max 582, avg 370
Indexing sequences 100%  
Sorting by abundance 100%
Counting unique k-mers 100%  
Detecting chimeras 0%[New Thread 0x7ffbbfba8700 (LWP 5698)]
Detecting chimeras 0%
Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffbbfba8700 (LWP 5698)]
dbindex_getbitmap (si=0x94c570) at dbindex.h:39
39	dbindex.h: No file of that type.
	in dbindex.h
Missing separate debuginfos, use: debuginfo-install glibc-2.12-1.149.el6_6.5.x86_64 libgcc-4.4.6-4.el6.x86_64 libstdc++-4.4.6-4.el6.x86_64
(gdb) backtrace 
#0  dbindex_getbitmap (si=0x94c570) at dbindex.h:39
#1  search_topscores (si=0x94c570) at searchcore.cc:119
#2  0x00000000004176e6 in search_onequery (si=0x94c570) at searchcore.cc:536
#3  0x000000000040ca77 in chimera_thread_core (ci=0x94c390) at chimera.cc:1132
#4  0x00000030382079d1 in start_thread () from /lib64/libpthread.so.0
#5  0x0000003037ae88fd in clone () from /lib64/libc.so.6
```

For the above command, I used the official binary. A fresh compilation gives a slightly different message:
```
8789864140 nt in 23735051 seqs, min 32, max 582, avg 370
Indexing sequences 100%  
Sorting by abundance 100%
Counting unique k-mers 100%  
Detecting chimeras 0%[New Thread 0x7ffbbfba8700 (LWP 17835)]
Detecting chimeras 0%
Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffbbfba8700 (LWP 17835)]
dbindex_getbitmap (si=0x92c560) at dbindex.h:39
39	    return kmerbitmap[kmer]->bitmap;
Missing separate debuginfos, use: debuginfo-install bzip2-libs-1.0.5-7.el6_0.x86_64 glibc-2.12-1.149.el6_6.5.x86_64 libgcc-4.4.6-4.el6.x86_64 libstdc++-4.4.6-4.el6.x86_64
(gdb) backtrace 
#0  dbindex_getbitmap (si=0x92c560) at dbindex.h:39
#1  search_topscores (si=0x92c560) at searchcore.cc:119
#2  0x0000000000417286 in search_onequery (si=0x92c560) at searchcore.cc:536
#3  0x000000000040c7f7 in chimera_thread_core (ci=0x92c380) at chimera.cc:1132
#4  0x00000030382079d1 in start_thread () from /lib64/libpthread.so.0
#5  0x0000003037ae88fd in clone () from /lib64/libc.so.6
```

Maybe the problem is coming from an integer overflow in searchcore.cc (line 116):
```
for(unsigned int i=0; i<si->kmersamplecount; i++) {
    unsigned int kmer = si->kmersample[i];
    unsigned char * bitmap = dbindex_getbitmap(kmer);
```

**Comment by torognes** [comment]
I think this should be fixed in version 1.1.2. Embarrassing bug that applied to searches in large datasets.

**Comment by torognes** [comment]
The problem is not fully resolved. The progress indicator is wrong.

**Comment by torognes** [comment]
Fully resolved in 1.1.3 I hope.
